press/agent.py

```diff
@@ -101,6 +103,21 @@ def reinstall_site(self, site):
 			site=site.name,
 		)
 
+	def check_database_file(self, site):
+		remote_database_file = site.remote_database_file
+		mem_database_file = requests.get(url = frappe.get_doc("Remote File", site.remote_database_file).download_link)
+		# get frappe version from the first line
+		frappe_version = re.search("Frappe (\d{1,2})\.\d{1,2}\.\d{1,2}", gzip.decompress(mem_database_file.content).decode("utf-8"))
```

Comment:
decompressing the backup file just to check version doesn't seem to optimal.
as @ankush pointed out it could be in GBs.

Maybe add a cleanup function?